src/common/constants.js

```diff
@@ -14,6 +14,7 @@ const LOCAL_PARSER_PORT = 8080;
 
```

Comment:
It feels like there's some new non-trivial business logic that needs to be under test with this change.

Reply:
Also as a refactor in the future, it would be nice to have GameMode become an object, and store that on the state, so that we can just call `gamemode.newGame();` and don't have to handle the business logic of detecting and handling different game modes at the call site.